File: sample/src/main/java/me/xiaopan/sketchsample/fragment/ImageOrientationTestFragment.java
package me.xiaopan.sketchsample.fragment;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;

import butterknife.BindView;
import me.xiaopan.sketch.SketchImageView;
import me.xiaopan.sketchsample.BaseFragment;
import me.xiaopan.sketchsample.BindContentView;
import me.xiaopan.sketchsample.R;

@BindContentView(R.layout.fragment_image_orientation_test)
public class ImageOrientationTestFragment extends BaseFragment {
    private static final String PARAM_REQUIRED_STRING_FILE_PATH = "PARAM_REQUIRED_STRING_FILE_PATH";

    @BindView(R.id.image_imageOrientationTestFragment_before)
    SketchImageView beforeImageView;

    @BindView(R.id.image_imageOrientationTestFragment_after)
    SketchImageView afterImageView;

    private String filePath;

    public static ImageOrientationTestFragment build(String filePath) {
        Bundle bundle = new Bundle();
        bundle.putString(PARAM_REQUIRED_STRING_FILE_PATH, filePath);

        ImageOrientationTestFragment fragment = new ImageOrientationTestFragment();
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Bundle arguments = getArguments();
        if (arguments != null) {
            filePath = arguments.getString(PARAM_REQUIRED_STRING_FILE_PATH);
        }

        if (TextUtils.isEmpty(filePath)) {
            throw new IllegalArgumentException("Not found filePath param");
        }
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        beforeImageView.getOptions().setCorrectImageOrientationDisabled(true);
        beforeImageView.displayImage(filePath);

        afterImageView.displayImage(filePath);
    }
}

File: sample/src/main/java/me/panpf/sketch/sample/fragment/ImageOrientationTestFragment.kt
package me.panpf.sketch.sample.fragment

import android.os.Bundle
import android.text.TextUtils
import android.view.View
import kotlinx.android.synthetic.main.fragment_image_orientation_test.*
import me.panpf.sketch.SketchImageView
import me.panpf.sketch.sample.BaseFragment
import me.panpf.sketch.sample.BindContentView
import me.panpf.sketch.sample.R

@BindContentView(R.layout.fragment_image_orientation_test)
class ImageOrientationTestFragment : BaseFragment() {

    val beforeImageView: SketchImageView by lazy {image_imageOrientationTestFragment_before}
    val afterImageView: SketchImageView by lazy {image_imageOrientationTestFragment_after}

    private var filePath: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val arguments = arguments
        if (arguments != null) {
            filePath = arguments.getString(PARAM_REQUIRED_STRING_FILE_PATH)
        }

        if (TextUtils.isEmpty(filePath)) {
            throw IllegalArgumentException("Not found filePath param")
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        beforeImageView.options.isCorrectImageOrientationDisabled = true
        beforeImageView.displayImage(filePath!!)

        TODO("不起作用了")
        TODO("往回滑动内容消失")

        afterImageView.displayImage(filePath!!)
    }

    companion object {
        private const val PARAM_REQUIRED_STRING_FILE_PATH = "PARAM_REQUIRED_STRING_FILE_PATH"

        fun build(filePath: String): ImageOrientationTestFragment {
            val bundle = Bundle()
            bundle.putString(PARAM_REQUIRED_STRING_FILE_PATH, filePath)

            val fragment = ImageOrientationTestFragment()
            fragment.arguments = bundle
            return fragment
        }
    }
}
